### PR TITLE
Alessandro/onboarding add to dock clean up old code

### PR DESCRIPTION
**Pull request by alessandroboron**
Task/Issue URL: https://app.asana.com/0/1206329551987282/1208258905717852/f

**Description**:

This PR removes the onboarding experiment branches and sets the onboarding highlights and contextual dialogs for all the users.

**Steps to test this PR**:
1. You can use maestro to make sure onboarding works as expected:
1.1 Run `./.maestro/setup_ui_tests.sh` to setup the simulator
1.2 Rune ./.maestro/run_ui_tests.sh onboarding_tests/01_onboarding.yaml to run the test.

**SCENARIO 1 - Onboarding Smoke Test**
Smoke test the onboarding and ensure that:
1. Linear onboarding presents App icon color selection and address bar position selection (iPad only app icon selection) 
2. Contextual dialogs are not presented more than once to the user.

**SCENARIO 2 - Add to Favorite Widget Flow**
I re-enabled the Add To Favorite flow from the widget. 

BEFORE:

https://github.com/user-attachments/assets/12e77f50-d24b-4d2a-8529-b90959129b96

AFTER:

https://github.com/user-attachments/assets/20e80c1e-be07-40a2-a54e-d7e3c29ba070

**SCENARIO 3 - Light Background iOS 15**
1. Ensure that light gradient renders fine on iPhone and iPad https://app.asana.com/0/1206329551987282/1208839072951158/f

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/OnboardingExperiment/Background/OnboardingGradient.swift [hunk 1]
```diff
@@ -23,24 +23,10 @@ import Onboarding
...
-        case (.default, .dark):
-            linearDarkGradient
-        case (.highlights, _):
-            // If highlights experiment use new common gradient for iOS and macOS
-            OnboardingGradient()
-        @unknown default:
-            linearLightGradient
-        }
+        // TODO: https://app.asana.com/0/1206329551987282/1208839072951158/f
+        // iOS 15 doesn't render properly the light EllipticalGradient
```
Comment (alessandroboron) [inline DuckDuckGo/OnboardingExperiment/Background/OnboardingGradient.swift]: Issue explained in https://app.asana.com/0/0/1208839072951158/1208895765286109/f

File: DuckDuckGo/OnboardingExperiment/Background/OnboardingGradient.swift [hunk 2]
```diff
@@ -23,57 +23,22 @@ import Onboarding
...
-            linearLightGradient
-        case (.default, .dark):
-            linearDarkGradient
-        case (.highlights, _):
-            // If highlights experiment use new common gradient for iOS and macOS
+        switch colorScheme {
+        case .dark:
             OnboardingGradient()
+        case .light:
+            // iOS 15 doesn't render properly the light EllipticalGradient while the Dark gradient is rendered correctly
```
Comment (alessandroboron) [inline DuckDuckGo/OnboardingExperiment/Background/OnboardingGradient.swift:31]: Issue explained in https://app.asana.com/0/0/1208839072951158/1208895765286109/f

File: DuckDuckGo/OnboardingExperiment/Background/OnboardingGradient.swift [hunk 3]
```diff
@@ -23,57 +23,22 @@ import Onboarding
...
+            // https://app.asana.com/0/1206329551987282/1208839072951158/f
+            if #available(iOS 16, *) {
+                OnboardingGradient()
+            } else {
+                Image(.onboardingGradientLight)
+                    .resizable()
```
Comment (brindy) [inline DuckDuckGo/OnboardingExperiment/Background/OnboardingGradient.swift:37]: nit: do we need resizable modifier if we're not changing the size?

**Review by brindy — APPROVED** [review]
Good job! one minor nit

File: DuckDuckGo/OnboardingExperiment/Background/OnboardingGradient.swift (hunk 3 above)
Comment (alessandroboron) [reply DuckDuckGo/OnboardingExperiment/Background/OnboardingGradient.swift:37]: on iPad I noticed it was rendered smaller than the screen size.

**Pull request merged**